### PR TITLE
Fix module circle reference issues and make prefab work

### PR DESCRIPTION
**Pull request by dumganhar**
Re: cocos-creator/fireball#

Changelog:
 *

### CODE REVIEW

File: cocos/core/components/deprecated.ts [hunk 1]
```diff
@@ -27,9 +27,9 @@
 /* eslint-disable @typescript-eslint/no-unsafe-return */
 
 import { Camera } from './camera-component';
-import { replaceProperty } from '../utils';
 import { legacyCC } from '../global-exports';
 import { js } from '../utils/js';
+import { replaceProperty } from '../utils/x-deprecated';
```
Comment (dumganhar) [inline cocos/core/components/deprecated.ts:32]: import 到子目录可以解决循环依赖问题

File: cocos/core/game.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@ import { legacyCC, VERSION } from './global-exports';
 import { IPhysicsConfig } from '../physics/framework/physics-config';
 import { bindingMappingInfo } from './pipeline/define';
 import { SplashScreen } from './splash-screen';
-import { RenderPipeline } from './pipeline';
+import { RenderPipeline } from './pipeline/render-pipeline';
```
Comment (minggo) [inline cocos/core/game.ts]: 这样引用有问题吧，替换后没有 rener-pipeline 类了。

Comment (dumganhar) [reply cocos/core/game.ts]: 额，奇怪，没出问题。改回去会有这个问题

 [Build] Circular dependency: ../engine/cocos/core/platform/index.ts -> ../engine/cocos/core/platform/deprecated.ts -> ../engine/cocos/core/utils/index.ts -> ../engine/cocos/core/utils/prefab-link.ts -> ../engine/cocos/core/assets/prefab.ts -> ../engine/cocos/core/utils/prefab/index.ts -> ../engine/cocos/core/utils/prefab/prefab-info.ts -> ../engine/cocos/core/components/index.ts -> ../engine/cocos/core/components/camera-component.ts -> ../engine/cocos/core/platform/view.ts -> ../engine/cocos/core/game.ts -> ../engine/cocos/core/scene-graph/index.ts -> ../engine/cocos/core/scene-graph/index.jsb.ts -> ../engine/cocos/core/scene-graph/scene.ts -> ../engine/cocos/core/scene-graph/scene.jsb.ts -> ../engine/cocos/core/platform/index.ts

Comment (dumganhar) [reply cocos/core/game.ts]: @minggo  Revert 了，改为修改 scene.jsb.ts 导入方式。

**Pull request merged**